src/Smalot/PdfParser/Document.php

```diff
@@ -169,6 +185,11 @@ public function getObjectById($id)
         return null;
     }
 
+    public function hasObjectsByType($type, $subtype = null)
```

Comment:
Since its a new function, please make it more explicit about what it expects and returns. Please check my suggestion and adapt accordingly.

```suggestion
    public function hasObjectsByType(string $type, ?string $subtype = null): bool
```